### PR TITLE
Refactor logic

### PR DESCRIPTION
**Pull request by victorbalan**

### CODE REVIEW

File: controllers/result.go [hunk 1]
```diff
@@ -204,8 +204,8 @@ func getLatestSubmissionKey(ctx context.Context, resultKey, taskKey *datastore.K
 	return keys[0], nil
 }
 
-func computeFinalScore(ctx context.Context, challenge model.Challenge, result model.Result, resultKey *datastore.Key) {
-	if err := logic.Resulter(challenge.Resulter).Call(ctx, resultKey); err != nil {
+func computeFinalScore(ctx context.Context, challenge model.Challenge, result model.KeyedResult) {
```
Comment (flowlo) [inline controllers/result.go]: The ordering of params is backwards. Would be nice to have the same order as for `logic.Resulter`

File: logic/logic.go [hunk 2]
```diff
@@ -15,14 +14,14 @@ type Resulter int
...
-type ResulterFunc func(ctx context.Context, resultKey *datastore.Key) error
+type ResulterFunc func(ctx context.Context, result model.KeyedResult, challenge model.Challenge) error
 
 // Tasker is a reference to a unique implementation of a TaskerFunc.
 type Tasker int
 
 // TaskerFunc is a function that will compute task results for the given Task
 // and User.
-type TaskerFunc func(ctx context.Context, task, result, user *datastore.Key) (model.Skills, error)
+type TaskerFunc func(ctx context.Context, task model.KeyedTask, result model.KeyedResult, user model.User, challenge model.Challenge) (model.Skills, error)
```
Comment (flowlo) [inline logic/logic.go]: I don't quite get why a `TaskerFunc` would need to access the challenge? They should be independent of a challenge, right?

Comment (victorbalan) [inline logic/logic.go]: we need the challenge to get the task index to know what time to get from the result.

Comment (flowlo) [inline logic/logic.go]: Then just pass the time to the tasker.

**Comment by flowlo** [comment]
LGTM

**Pull request merged**